tests/integration/cqlengine/connections/test_connection.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2013-2016 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    import unittest2 as unittest
+except ImportError:
+    import unittest  # noqa
+
+
+from cassandra.cqlengine.models import Model
+from cassandra.cqlengine import columns, connection
+from cassandra.cqlengine.management import sync_table
+from cassandra.cluster import Cluster
+from cassandra.query import dict_factory
+
+from tests.integration import PROTOCOL_VERSION, execute_with_long_wait_retry
+from tests.integration.cqlengine.base import BaseCassEngTestCase
+
+
+class TestConnectModel(Model):
+
+    id = columns.Integer(primary_key=True)
+    keyspace = columns.Text()
+
+
+class ConnectionTest(BaseCassEngTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.keyspace1 = 'ctest1'
+        cls.keyspace2 = 'ctest2'
+        super(ConnectionTest, cls).setUpClass()
+        cls.setup_cluster = Cluster(protocol_version=PROTOCOL_VERSION)
+        cls.setup_session = cls.setup_cluster.connect()
+        ddl = "CREATE KEYSPACE {0} WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': '{1}'}}".format(cls.keyspace1, 1)
+        execute_with_long_wait_retry(cls.setup_session, ddl)
+        ddl = "CREATE KEYSPACE {0} WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': '{1}'}}".format(cls.keyspace2, 1)
+        execute_with_long_wait_retry(cls.setup_session, ddl)
+
+    @classmethod
+    def tearDownClass(cls):
+        execute_with_long_wait_retry(cls.setup_session, "DROP KEYSPACE {0}".format(cls.keyspace1))
+        execute_with_long_wait_retry(cls.setup_session, "DROP KEYSPACE {0}".format(cls.keyspace2))
+        cls.setup_cluster.shutdown()
+
+    def setUp(self):
+        self.c = Cluster(protocol_version=PROTOCOL_VERSION)
+        self.session1 = self.c.connect(keyspace=self.keyspace1)
+        self.session1.row_factory = dict_factory
+        self.session2 = self.c.connect(keyspace=self.keyspace2)
+        self.session2.row_factory = dict_factory
+
+    def test_connection_session_switch(self):
```

Comment:
I think we might leak a few sessions here: session1, session2, and the one that is originally setup in cqlengine.connection. Should we shut those down?